typescript/provider-sdk/src/artifact.ts

```diff
@@ -27,6 +28,21 @@ export type ArtifactDeployed<C, D> = {
   deployed: D;
 };
 
+/**
+ * Represents an artifact that has been deployed on chain
+ * but is represented only by its address
+ */
+export type ArtifactUnderived = {
+  artifactState: typeof ArtifactState.UNDERIVED;
+  artifactAddress: string;
```

Comment:
As discussed elsewhere, there may be good reasons to use the derived type `<D>` here to accommodate situations when new contracts are represented by more than a single string address.

Reply:
Yup just updated to use the `D` param